webapp/src/shared/components/templates/RowsPageTemplate/RowsPageTemplate.tsx

```diff
@@ -0,0 +1,82 @@
+import './RowsPageTemplate.css';
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { USER_URL, AVATAR_EP_URL, WILDCARD_AVATAR_URL } from '../../../urls';
+import { MediumAvatar } from '../../Avatar/Avatar';
+import SearchForm from '../../Input/SearchForm';
+import RowsList, { RowItem } from '../../RowsList/RowsList';
+import NavigationBar from '../../NavigationBar/NavigationBar';
+import { useData } from '../../../hooks/UseData';
+import Loading from '../../Loading/Loading';
+import { usersApi } from '../../../services/ApiService';
+import { useSearchContext } from '../../../context/SearchContext';
+
+type RowsPageTemplateProps<T> = {
+  dataValidator: (data: object) => boolean;
+  dataMapper: (data: T) => RowItem;
+};
```

Comment:
I added the `button ?: JSX.Element` to what we used to call DispatchPage Props because even back then we could already see that we may need a button below the list of rows, maybe not for the users page, but certainly for the chat. Why did you decide to remove it, now that it looks like we may actually need this feature?

Reply:
I agree with @Damrod, or at least I was using this button feature in the chat page.
Or we can add it separately afterwards

Reply:
I'd say to add later it in chat's page task, we will know better the needs we have there!